### ISSUE TITLE
[crd-gen] Support openAPIV3Schema fields: minimum, maximum, pattern, nullable

### ISSUE THREAD

**Issue by leoandrea7** [fabric8io/kubernetes-client#4224]
### Is your enhancement related to a problem? Please describe

_(duplicate of #2959)_

At the moment there is no way to auto generate a crd (with crd-gen tool) that contains properties with validation fields like **_minimum_**, **_maximum_**, **_pattern_**, **_nullable_**.

CRD example:

```yaml
...

apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: tenants.example.com
spec:
  group: example.com
  names:
    kind: Tenant
    plural: tenants
    singular: tenant
  scope: Namespaced
  versions:
  - name: v1
    schema:
      openAPIV3Schema:
        properties:
          spec:
            properties:
              kafka:
                type: object
                properties:
                  port:
                    type: integer
                    minimum: 0
                    maximum: 65535
                    nullable: false
                  hostname:
                    type: string
                    pattern: '^([A-Za-z0-9-]{1,63}\.)+[[A-Za-z0-9-]{1,63}$'
                    nullable: false
                nullable: false
...
```

### Describe the solution you'd like

I'd like to use javax annotation (like `@min @max @pattern`) on a java model fields.

    private String protocol;

    @Pattern(regexp = "\\^\\(\\[A-Za-z0-9-]\\{1,63}\\\\\\.\\)\\+\\[\\[A-Za-z0-9-]\\{1,63}\\$")
    private String hostname;

    @Min(0)
    @Max(65535)
    private Integer port;

Currently @NotNull annotation generate the filed "**_required_**" on crd, but for "_**nullable**_" validation maybe we need of a new specific annotation.

### Additional context

version: 5.12.2

**Comment by OneCricketeer** [comment]
Is the pattern in your model example correct? The `^` and parentheses should not be literals, for example. The only change from the yaml should be escaping the one `\\.`

**Comment by leoandrea7** [comment]
Yes you are right but is not important the regex value now...The problem here is that crd-gen not have the featuare to read `@Pattern @Max @Min` and to generate corresponding fields(pattern, maximum and minimum) into the generated crd yaml file.
I think that the logic to handle this annotations should be into [AbstractJsonSchema](https://github.com/fabric8io/kubernetes-client/blob/master/crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java) as for the other already supported annotations.

**Comment by sunix** [comment]
@andreaTP could you have a look at this issue?

**Comment by andreaTP** [comment]
I'm about to go on vacation, but if no one picks it up earlier I might eventually take a look at this.

**Comment by manusa** [comment]
Relates to:
- https://github.com/fabric8io/kubernetes-client/issues/2611#issuecomment-745088459
- https://github.com/fabric8io/kubernetes-client/issues/2959 (duplicate)

**Comment by metacosm** [comment]
I have to admit that I'm a little lost on the subtleties of `required` and `nullable`… if something is not marked as `required`, shouldn't it be `nullable` by default? Conversely, if something is marked `required`, it cannot be `nullable` so I'm not sure why we need to support both… especially because then there might be conflicts i.e. what is the generator supposed to do (apart from throwing an error, I guess) if a property is both marked as `nullable` and `required`?

**Comment by andreaTP** [comment]
Closed in #4348
